File: lua_scripts/config/server_config.lua
-- 服务器配置

local M = {
    -- 服务器信息
    server = {
        name = "LegendServer",
        version = "1.0.0",
    },

    -- 网络配置
    network = {
        gateway_port = 8888,
        game_port = 8889,
        max_connections = 10000,
        timeout = 90000,  -- 心跳超时(ms)
    },

    -- 数据库配置
    database = {
        mysql = {
            host = "127.0.0.1",
            port = 3306,
            database = "legend_db",
            user = "legend",
            password = "legend123",
            pool_size = 20,
        },
        redis = {
            host = "127.0.0.1",
            port = 6379,
            db = 0,
            pool_size = 10,
        },
    },

    -- 游戏配置
    game = {
        -- 角色配置
        role = {
            max_name_length = 16,
            max_per_account = 3,
            default_hp = 100,
            default_mp = 100,
        },

        -- 战斗配置
        battle = {
            max_rounds = 30,
            skill_cd_base = 1000,
        },

        -- 聊天配置
        chat = {
            max_length = 256,
            world_cd = 5000,
            private_cd = 1000,
        },

        -- 好友配置
        friend = {
            max_count = 100,
        },

        -- 组队配置
        team = {
            max_members = 5,
        },
    },

    -- 日志配置
    log = {
        level = "info",
        path = "./logs",
        max_size = 104857600,  -- 100MB
        max_files = 10,
    },
}

return M

File: src/core/server/server_config.cpp
/**
 * @file server_config.cpp
 * @brief 服务器配置实现
 */

#include "server_config.h"
#include <fstream>

namespace legend {

bool ServerConfig::load(const String& filename) {
    try {
        YAML::Node config = YAML::LoadFile(filename);

        // 服务器配置
        if (config["server"]) {
            auto server = config["server"];
            if (server["name"]) {
                serverName_ = server["name"].as<String>();
            }
            if (server["version"]) {
                serverVersion_ = server["version"].as<String>();
            }
        }

        // 网关配置
        if (config["server"] && config["server"]["gateway"]) {
            auto gateway = config["server"]["gateway"];
            if (gateway["host"]) {
                gatewayHost_ = gateway["host"].as<String>();
            }
            if (gateway["port"]) {
                gatewayPort_ = gateway["port"].as<uint16>();
            }
            if (gateway["max_connections"]) {
                maxConnections_ = gateway["max_connections"].as<int>();
            }
        }

        // 游戏服务器配置
        if (config["server"] && config["server"]["game"]) {
            auto game = config["server"]["game"];
            if (game["host"]) {
                gameHost_ = game["host"].as<String>();
            }
            if (game["port"]) {
                gamePort_ = game["port"].as<uint16>();
            }
            if (game["max_players"]) {
                maxPlayers_ = game["max_players"].as<int>();
            }
        }

        // 线程配置
        if (config["server"] && config["server"]["threads"]) {
            auto threads = config["server"]["threads"];
            if (threads["io_threads"]) {
                ioThreads_ = threads["io_threads"].as<int>();
            }
            if (threads["logic_threads"]) {
                logicThreads_ = threads["logic_threads"].as<int>();
            }
            if (threads["db_threads"]) {
                dbThreads_ = threads["db_threads"].as<int>();
            }
        }

        // 数据库配置
        if (config["database"] && config["database"]["mysql"]) {
            auto mysql = config["database"]["mysql"];
            if (mysql["host"]) {
                mysqlHost_ = mysql["host"].as<String>();
            }
            if (mysql["port"]) {
                mysqlPort_ = mysql["port"].as<uint16>();
            }
            if (mysql["user"]) {
                mysqlUser_ = mysql["user"].as<String>();
            }
            if (mysql["password"]) {
                mysqlPassword_ = mysql["password"].as<String>();
            }
            if (mysql["database"]) {
                mysqlDatabase_ = mysql["database"].as<String>();
            }
            if (mysql["pool_size"]) {
                mysqlPoolSize_ = mysql["pool_size"].as<int>();
            }
        }

        if (config["database"] && config["database"]["redis"]) {
            auto redis = config["database"]["redis"];
            if (redis["host"]) {
                redisHost_ = redis["host"].as<String>();
            }
            if (redis["port"]) {
                redisPort_ = redis["port"].as<uint16>();
            }
            if (redis["password"]) {
                redisPassword_ = redis["password"].as<String>();
            }
            if (redis["db"]) {
                redisDb_ = redis["db"].as<int>();
            }
            if (redis["pool_size"]) {
                redisPoolSize_ = redis["pool_size"].as<int>();
            }
        }

        // 日志配置
        if (config["server"] && config["server"]["log"]) {
            auto log = config["server"]["log"];
            if (log["level"]) {
                logLevel_ = log["level"].as<String>();
            }
            if (log["path"]) {
                logPath_ = log["path"].as<String>();
            }
        }

        // 心跳配置
        if (config["heartbeat"]) {
            auto heartbeat = config["heartbeat"];
            if (heartbeat["interval"]) {
                heartbeatInterval_ = heartbeat["interval"].as<int>();
            }
            if (heartbeat["timeout"]) {
                heartbeatTimeout_ = heartbeat["timeout"].as<int>();
            }
        }

        return true;

    } catch (const YAML::Exception& e) {
        return false;
    }
}

} // namespace legend